manage_rbac.py

```diff
@@ -174,43 +199,83 @@ def __init__(self, rbac_users={}):
             logging.info("No Role Bindings need to be deleted")
         else:
             logging.info("Deleting Role Bindings")
             for rb in role_bindings_to_delete:
                 logging.info("Deleting Role Binding: {} in {} namespace".format(rb.metadata.name, rb.metadata.namespace))
-                rbac_client.delete_namespaced_role_binding(
-                  namespace=rb.metadata.namespace,
-                  name=rb.metadata.name,
-                  body=kubernetes.client.V1DeleteOptions(),
-                  pretty=True
-                )
+                try:
+                    rbac_client.delete_namespaced_role_binding(
+                      namespace=rb.metadata.namespace,
+                      name=rb.metadata.name,
+                      body=kubernetes.client.V1DeleteOptions(),
+                      pretty=True
+                    )
+                except ApiException, e:
+                    logging.error("Error deleting Role Binding: {} in {} namespace".format(rb.metadata.name, rb.metadata.namespace))
+                    logging.error(e)
+
+    def __connect(self):
+        """ Connect to the kubernetes API """
+
+        logging.debug("Connecting to Kubernetes API")
+        k8s_config_loaded = False
+
+        try:
+            logging.debug("Attempting to load incluster config")
+            kubernetes.config.load_incluster_config()
+            logging.debug("Successfully loaded incluster config")
+        except ConfigException:
+            logging.debug("Loading incluster config failed")
+            logging.debug("Attempting to load kube config")
+            try:
+                kubernetes.config.load_kube_config()
+                logging.debug("Successfully loaded kube config")
+            except Exception, e:
+                logging.error("Loading kube config failed, exiting")
+                raise RBACManagerException("Error connecting to kubernetes: {}".format(e))
 
-        logging.info("---")
+        self._k8s_client = kubernetes.client
+        logging.debug(self._k8s_client)
+
+    @property
+    def k8s_client(self):
+        try:
+            getattr(self, "_k8s_client")
+        except AttributeError:
+            self.__connect()
+        return self._k8s_client
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='Updates RBAC cluster role bindings and role bindings.')
-    parser.add_argument('--config', help='YAML configuration file to load', required=True)
+    parser.add_argument('--config', help='YAML configuration file to load')
     parser.add_argument('--kubectl-auth', action='store_true', help='Use kubectl command to refresh auth (useful for GKE)')
     args = parser.parse_args()
     if args.kubectl_auth:
         os.system('kubectl get ns >/dev/null 2>&1')
-    RBACManager(yaml.load(open(args.config)))
+    try:
+        if args.config is not None:
+            RBACManager().update(file=args.config)
```

Comment:
It seems like some logging here and on the line below to explain what mode this is running in would be helpful.

Reply:
👍